lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -587,6 +588,36 @@ instance NFData SealedTx where
         -- Showing the transaction should be enough to fully evaluate it.
         tx' = if v then show tx else ""
 
+-- Helper function to constrain the era of a 'SealedTx' to at most the provided
+-- era. If this is not possible, the original 'SealedTx' is returned.
+--
+-- In contrast to the "most recent era" argument of @sealedTxFromBytes'@, this
+-- function allows constraining the era at a point after the tx has been
+-- deserialised. For instance, in a server handler with known current era,
+-- instead of in an @Aeson.FromJSON@ instance.
+--
+-- >>> ideallyNoLaterThan alonzoEra alonzoCompatibleBabbageTx
+-- alonzoCompatibleBabbageTx
+--
+-- >>> ideallyNoLaterThan alonzoEra alonzoIncompatibleBabbageTx
+-- babbageTx
+--
+-- == Is this what we want? (for the new tx-workflow..)
+--
+-- Probably not. This is a minimally invasive approach to ensure:
+-- - tx workflow works in both Alonzo and Babbage
+-- - tx workflow tries to create Alonzo txs in Alonzo and Babbage txs in Babbage
+--
+-- With the added behaviour:
+-- - tx workflow may partially work for babbage-only txs when in alonzo
```

Comment:
🤔

Reply:
🤔 I agree. We may need to rethink this at some later point.

I have the feeling that the preferred use `cardanoTxIdeallyNoLaterThan` makes the `unsafeCardanoTx` field obsolete. (We could make the field obsolete now if it were not for functions such as `getSealedTxBody` and `getSealedTxWitnesses`.) I'll add more comments to ADP-2044.